composer.json

```diff
@@ -5,7 +5,7 @@
     "license": "MIT",
     "require": {
         "php" : ">=5.3.0",
-        "swiftmailer/swiftmailer": "~5.2"
+        "swiftmailer/swiftmailer": "~5.2|~6.0"
```

Comment:
The dependency constraint should be more strict like that => `^6.0`.

Reply:
Thanks, could you please explain the difference to me? From what I understood on https://getcomposer.org/doc/articles/versions.md#next-significant-release-operators, it will be the same and only differ, if I would introduce the third digit.

Reply:
The `~` include `7.0@dev` but not with `^`.

Reply:
> The `~` include `7.0@dev` but not with `^`.

Hm... Really? 

I think they explicitly mention that it will not install versions having major number different from `6` @alquerci 

> Note: Although `2.0-beta.1` is strictly before `2.0`, a version constraint like `~1.2` would not install it. As said above `~1.2` only means the `.2` can change but the `1.` part is fixed.

https://getcomposer.org/doc/articles/versions.md#tilde-version-range-

Reply:
Hey, indeed I wasn't understand in details the difference.

Now yes.

> The ^ operator behaves very similarly, but it sticks closer to semantic versioning, and will always allow non-breaking updates.

https://getcomposer.org/doc/articles/versions.md#caret-version-range-

#learningAllDays